src/foam/nanos/cron/CronScheduler.java

```diff
@@ -36,31 +38,33 @@ private Date getMinScheduledTime() {
     return ((Cron) min.getValue()).getScheduledTime();
   }
 
-  private final Thread cronJobs = new Thread() {
-    @Override
-    public void run() {
-      try {
-        while (true) {
-          Date dtnow = new Date();
-          cronDAO_.where(MLang.LTE(Cron.SCHEDULED_TIME, dtnow)).select(new AbstractSink() {
-            @Override
-            public void put(FObject obj, Detachable sub) {
-              ((Cron) obj).runScript(getX());
-              cronDAO_.put(obj);
-            }
-          });
+  public void start() {
+    cronDAO_ = (DAO) getX().get("cronDAO");
+    new Thread(this).start();
+  }
+
+  @Override
+  public void run() {
+    NanoLogger logger = (NanoLogger) getX().get("logger");
+    final PM pm = new PM(this.getClass(), "cronScheduler");
```

Comment:
Would I need to change the serviceKey in this case? Maybe to be something that represents the cron job being run instead of the cron scheduler as a whole.